portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java

```diff
@@ -38,24 +45,32 @@
 
   @Override  
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("comClass").addSort("timestamp", SortDirection.DESCENDING);
+    response.setContentType("text/html;");
+    PrintWriter out = response.getWriter();
+    
+    UserService userService = UserServiceFactory.getUserService();
+    List<Task> comments = new ArrayList<>();
+    if (userService.isUserLoggedIn()) {
```

Comment:
Very minor, but from a stylistic perspective, you'll typically do the reverse:

if (!userService.isUserLogginedIn()) {
    return;
}

This moves all your error handling logic up top, and also leaves your code cleaner because you don't have a big nested block once you've checked login.